learningequality/studio
# GlobalSnackbar using Vuetify for now

## Description

Copy-pastes the Kolibri `GlobalSnackbar` component and `snackbar` Vuex modules into Studio and tweaks a bit.

Rather than using the `CoreSnackbar` from Kolibri, this simply uses the Vuetify `v-snackbar` component.

This does not insert the `GlobalSnackbar` into every possibly root level context for now. Considering that we are using Vuetify for wrapper components - I suspect that @indirectlylit and I will find a way to ensure the component is available everywhere. I'm not sure now but I imagine that we will replace `VApp` with `CoreBase` or similar - in which case we can inject the `GlobalSnackbar` there.

**NOTE** - Once this is tested I will remove the changes that add the "Show snackbar" button. This is just for testing - I did it first in the sandbox but wanted to be sure it worked outside of the sandbox as well and it all looks good.

## Steps to Test

Copy paste this into `Sandbox.vue` then check out the sandbox. This implements the new action dispatch. You can test options this way as well as needed.

```
<template>

  <div>
    <a href="javascript:void(0);" @click="$store.dispatch('showSnackbar', { actionText: 'nahhh', duration: duration, text: 'hi!' })">Auto dismiss</a>
    <label>Enter duration in ms (0 if you want it to live forever)</label>
    <br>
    <input v-model="duration" type="text" style="border: 1px solid black;">
    <TemplateComponent />
    <GlobalSnackbar />
  </div>

</template>
<script>

  import TemplateComponent from '../../channelEdit/views/template/Template';
  import GlobalSnackbar from './GlobalSnackbar';

  export default {
    name: 'Sandbox',
    components: {
      TemplateComponent,
      GlobalSnackbar,
    },
    data() {
      return {
        duration: 1000,
      };
    },
  };

</script>


<style lang="less" scoped>

</style>
```

## Implementation Notes (optional)

#### At a high level, how did you implement this?

There is a Vuex module in the frontend/shared/modules folder that manages the state of the snackbar.

There is a `GlobalSnackbar` component which should be added to the root component of whatever you're using snackbars in (ie, `<GlobalSnackbar />`)

Dispatch the `showSnackbar` and `clearSnackbar` actions as needed. Pass an object like:

```
{
    text: <String>, // This is the message shown in the snackbar
    actionText: <String>, // This is what goes on the button that dismisses the snackbar
    duration: <Int / Null>, // Time in ms before snackbar auto-hides itself. null won't autohide
    actionCallback: <Function>, // Function to call when the snackbar action is clicked
}
```

~Commit the mutations to use it:~

~`CORE_CREATE_SNACKBAR`~
~`CORE_CLEAR_SNACKBAR`~
~`CORE_SET_SNACKBAR_TEXT`~

See what items you can pass to the snackbar here https://github.com/learningequality/kolibri/blob/fa14258f1d32c45b6e739e77a5d0676d20fe9e8f/kolibri/core/assets/src/state/modules/snackbar.js#L24 

#### Does this introduce any tech-debt items?

@indirectlylit and I will need to continue the migration to the new design system.

## Checklist

*Delete any items that don't apply*


## Comments

*Any additional notes you'd like to add*

## Reviewers

If you are looking to assign a reviewer, here are some options:
- Jordan jayoshih (full stack)
- Aron aronasorman (back end, devops)
- Micah micahscopes (full stack)
- Kevin kollivier (back end)
- Ivan ivanistheone ([Ricecooker](https://github.com/learningequality/ricecooker))
- Richard rtibbles (full stack, [Kolibri](https://github.com/learningequality/kolibri))
- Radina @radinamatic (documentation)
